### PR TITLE
Handle no response when looking up a station close to a user. Return …

### PR DESCRIPTION
**Pull request by ryancoughlin**
…empty `{}`

### CODE REVIEW

File: lib/geo.js [hunk 1]
```diff
@@ -26,7 +26,12 @@ export function findStation(longitude, latitude) {
         if (error) {
           reject(error);
         }
-        resolve(station.stationId);
+
+        if (station) {
+          resolve(station.stationId);
```
Comment (ryancoughlin) [inline lib/geo.js]: This error handling on checks data from Mongo, which is the biggest bottleneck now.

File: lib/geo.js [hunk 2]
```diff
@@ -26,7 +26,12 @@ export function findStation(longitude, latitude) {
...
+        } else {
+          reject("No station found");
```
Comment (ryancoughlin) [inline lib/geo.js]: Thoughts on this usage? I don't have an error to pass it, I don't really need the string. I want it to reject it and handle that appropriately. Would I do: `reject()` instead?

Comment (ryancoughlin) [inline lib/geo.js]: I can also do the same for forecast.io

Comment (BlakeWilliams) [inline lib/geo.js]: I'd stick with the error message, that way later on you can `alert(e)` and see where it failed

Comment (ryancoughlin) [inline lib/geo.js]: In this case, I don't get an error. It's just calling `reject` which gets bubbled back up to my promise

Comment (ryancoughlin) [inline lib/geo.js]: It is only triggered when `station` does not exist aka there are no results from the geoquery.

Comment (ryancoughlin) [inline lib/geo.js]: Right now if a user is in a location where they are too far away from a station, the server will crash. This is my way of trying to catch that and handle it gracefully.  💃

Comment (BlakeWilliams) [inline lib/geo.js]: Where in the code causes the crash?

Comment (ryancoughlin) [inline lib/geo.js]: When it passes that value back from the request. Error below

`TypeError: Cannot read property 'stationId' of null` - geo.js / line 29

https://github.com/ryancoughlin/Salty-Server/blob/master/lib/geo.js#L27

Comment (BlakeWilliams) [inline lib/geo.js]: I think your conditional fixed it, yeah?

**Pull request merged**